### PR TITLE
[SYCL][FPGA] Add support for [[intel::loop_count()]] attribute

### PR DESCRIPTION
**Pull request by smanna12**
The original implementation of this attribute in
https://github.com/intel/llvm/pull/3438, did not support
the [[intel::loop_count()]] attribute spelling. This patch
adds support for that spelling.

Signed-off-by: Soumi Manna <soumi.manna@intel.com>

### CODE REVIEW

**Comment by smanna12** [comment]
NOTE: I noticed this missing attribute spelling while working on unifying downstream source codes with pragma support.

File: clang/include/clang/Basic/Attr.td [hunk 1]
```diff
@@ -2058,12 +2058,14 @@ def : MutualExclusions<[SYCLIntelFPGAMaxConcurrency,
...
-                   CXX11<"intel", "loop_count_avg">];
+                   CXX11<"intel", "loop_count_avg">,
+                    CXX11<"intel", "loop_count">];
   let Subjects = SubjectList<[ForStmt, CXXForRangeStmt, WhileStmt, DoStmt],
                              ErrorDiag, "'for', 'while', and 'do' statements">;
   let Accessors = [Accessor<"isMin", [CXX11<"intel", "loop_count_min">]>,
                    Accessor<"isMax", [CXX11<"intel", "loop_count_max">]>,
-                   Accessor<"isAvg", [CXX11<"intel", "loop_count_avg">]>];
+                   Accessor<"isAvg", [CXX11<"intel", "loop_count_avg">]>,
+                   Accessor<"isCount", [CXX11<"intel", "loop_count">]>];
```
Comment (elizabethandrews) [inline clang/include/clang/Basic/Attr.td:2068]: Please update documentation as well

File: clang/include/clang/Basic/Attr.td [hunk 2]
```diff
@@ -2058,12 +2058,14 @@ def : MutualExclusions<[SYCLIntelFPGAMaxConcurrency,
 def SYCLIntelFPGALoopCount : StmtAttr {
   let Spellings = [CXX11<"intel", "loop_count_min">,
                    CXX11<"intel", "loop_count_max">,
-                   CXX11<"intel", "loop_count_avg">];
+                   CXX11<"intel", "loop_count_avg">,
+                    CXX11<"intel", "loop_count">];
```
Comment (zahiraam) [inline clang/include/clang/Basic/Attr.td:2062]: Formatting?

File: clang/include/clang/Basic/Attr.td (hunk 1 above)
Comment (smanna12) [reply clang/include/clang/Basic/Attr.td:2068]: Done

File: clang/include/clang/Basic/Attr.td (hunk 2 above)
Comment (smanna12) [reply clang/include/clang/Basic/Attr.td]: Done

**Review by mendell27 — APPROVED** [review]

**Comment by premanandrao** [comment]
> > NOTE: I noticed this missing attribute spelling while working on unifying downstream source codes with pragma support
> 
> Please verify this spelling and functionality is required for SYCL attribute before merging this. If SYCL backends don't know how to consume this, I am not sure if this required. @premanandrao @zahiraam please take a look

This is implementing
https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compiler-reference/pragmas/intel-specific-pragma-reference/loop-count.html
is it not, @smanna12?

@elizabethandrews, is this what you wanted verified?

**Comment by elizabethandrews** [comment]
> > > NOTE: I noticed this missing attribute spelling while working on unifying downstream source codes with pragma support
> > 
> > 
> > Please verify this spelling and functionality is required for SYCL attribute before merging this. If SYCL backends don't know how to consume this, I am not sure if this required. @premanandrao @zahiraam please take a look
> 
> This is implementing https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compiler-reference/pragmas/intel-specific-pragma-reference/loop-count.html is it not, @smanna12?
> 
> @elizabethandrews, is this what you wanted verified?

I wasn't sure if the solution was to add the `loop_count ` spelling here, or remove the equivalent downstream.

**Comment by smanna12** [comment]
> > > > NOTE: I noticed this missing attribute spelling while working on unifying downstream source codes with pragma support
> > > 
> > > 
> > > Please verify this spelling and functionality is required for SYCL attribute before merging this. If SYCL backends don't know how to consume this, I am not sure if this required. @premanandrao @zahiraam please take a look
> > 
> > 
> > This is implementing https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compiler-reference/pragmas/intel-specific-pragma-reference/loop-count.html is it not, @smanna12?
> > @elizabethandrews, is this what you wanted verified?
> 
> I wasn't sure if the solution was to add the `loop_count ` spelling here, or remove the equivalent downstream.

>>This is implementing
https://www.intel.com/content/www/us/en/develop/documentation/oneapi-dpcpp-cpp-compiler-dev-guide-and-reference/top/compiler-reference/pragmas/intel-specific-pragma-reference/loop-count.html
is it not, @smanna12?
Yes, pragma syntax support the "loop_count" spelling, so this adds the same spelling loop_count attribute now.

**Pull request merged**